ui/src/app/zones/views/ZonesListTable/ZonesListTable.tsx

```diff
@@ -0,0 +1,81 @@
+import { useEffect } from "react";
+
+import { MainTable } from "@canonical/react-components";
+import { useSelector, useDispatch } from "react-redux";
+import { Link } from "react-router-dom";
+
+import machineURLs from "app/machines/urls";
+import { actions } from "app/store/zone";
+import zoneSelectors from "app/store/zone/selectors";
+
+const ZonesListTable = (): JSX.Element => {
+  const zones = useSelector(zoneSelectors.all);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(actions.fetch());
+  }, [dispatch]);
+
+  const headers = [
+    { content: "Name", sortKey: "name" },
+    { content: "Description", sortKey: "description" },
+    { content: "Machines", sortKey: "machines", className: "u-align--right" },
+    { content: "Devices", sortKey: "devices", className: "u-align--right" },
+    {
+      content: "Controllers",
+      sortKey: "controllers",
+      className: "u-align--right",
+    },
+  ];
+  const rows = zones.map((zone) => {
+    return {
```

Comment:
A small convenience with arrow functions is that you can return directly by doing `const rows = zones.map((zone) => ({...`.

Note that you need `()` around the object otherwise it assumes the curlies are wrapping the function content.